src/Compilers/CSharp/Test/Semantic/Semantics/ConstantTests.cs

```diff
@@ -2360,8 +2360,7 @@ void M()
             // Duplicate "(byte)2" is because there's an implicit conversion to uint.
             // Duplicate "b ? (uint)1 : (byte)2" is because there's an implicit conversion to int.
             var expected =
-@"true ? 1 + i : (int)4u --> BAD
```

Comment:
>"true ? 1 + i : (int)4u --> BAD [](start = 1, length = 31)

Why do we want this behavior change? All the parts are constants, so I'd expect the expression to still have a (bad) constant value.

Reply:
> Why do we want this behavior change?

The previous behavior (copying the ConstantValue from one of the branches of the conditional operator when the condition is a constant and the branch is ConstantValue.Bad) was an artifact of an earlier optimization when the branches were identical types (see [Binder_Operators.cs](https://github.com/dotnet/roslyn/blob/27ede702ea49770fcb551e3d2e571dc51ff4b319/src/Compilers/CSharp/Portable/Binder/Binder_Operators.cs#L3752)). If the branches were not identical, the constant value was not copied.

That meant the constant value was set on the first `?:` expression below but not the second. Since the behavior was inconsistent between the two cases, the optimization for identical types did not seem necessary.
```
const int i = i;
const int z = true ? 1 + i : (int)4u;
const int y = true ? 1 + i : 4u;
```